### ISSUE TITLE
[Fatal Error] Markup in flutter_embedding_debug POM is not well-formed

### ISSUE THREAD

**Issue by majid-2002** [flutter/flutter#159927]
### Steps to reproduce

1. Create a new Flutter project.
2. Run the project with flutter run.

### Expected results

Should run without any error.

### Actual results

When attempting to build and run the Flutter project using flutter run, the build fails with the following error message:

```console
[Fatal Error] flutter_embedding_debug-1.0.0-a18df97ca57a249df5d8d68cd0820600223ce262.pom:2:90: The markup in the document following the root element must be well-formed.

```

The error seems to originate from the flutter_embedding_debug POM file.
The build process completes when running but the error is shown .
The issue persists across multiple attempts, even after ensuring dependencies in pubspec.yaml are up-to-date.

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

![image](https://github.com/user-attachments/assets/97819895-f5da-4014-be92-ee97026065b4)

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Fatal Error] flutter_embedding_debug-1.0.0-a18df97ca57a249df5d8d68cd0820600223ce262.pom:2:90: The markup in the document following the root element must be well-formed.

```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
flutter doctor -v
[√] Flutter (Channel stable, 3.24.5, on Microsoft Windows     
    [Version 10.0.19045.5198], locale en-IN)
    • Flutter version 3.24.5 on channel stable at C:\flutter  
    • Upstream repository
      https://github.com/flutter/flutter.git
    • Framework revision dec2ee5c1f (3 weeks ago), 2024-11-13 
      11:13:06 -0800
    • Engine revision a18df97ca5
    • Dart version 3.5.4
    • DevTools version 2.37.3

[√] Windows Version (Installed version of Windows is version  
    10 or higher)

Checking Android licenses is taking an unexpectedly long time.

[√] Android toolchain - develop for Android devices (Android
    SDK version 34.0.0)
    • Android SDK at
      C:\Users\abdulmajid\AppData\Local\Android\sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\Program Files\Android\Android      
      Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build       
      21.0.3+-12282718-b509.11)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program
      Files\Google\Chrome\Application\chrome.exe

[!] Visual Studio - develop Windows apps (Visual Studio 
    Community 2022 17.11.5)
    • Visual Studio at C:\Program Files\Microsoft Visual
      Studio\2022\Community
    • Visual Studio Community 2022 version 17.11.35327.3      
    X Visual Studio is missing necessary components. Please   
      re-run the Visual Studio installer for the "Desktop     
      development with C++" workload, and include these       
      components:
        MSVC v142 - VS 2019 C++ x64/x86 build tools
         - If there are multiple build tool versions
         available, install the latest
        C++ CMake tools for Windows
        Windows 10 SDK

[√] Android Studio (version 2024.2)
    • Android Studio at C:\Program Files\Android\Android      
      Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter      
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      21.0.3+-12282718-b509.11)

[√] VS Code (version 1.95.3)
    • VS Code at
      C:\Users\abdulmajid\AppData\Local\Programs\Microsoft VS 
      Code
    • Flutter extension version 3.102.0

[√] Connected device (4 available)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 •
      android-x64    • Android 15 (API 35) (emulator)
    • Windows (desktop)            • windows       •
      windows-x64    • Microsoft Windows [Version
      10.0.19045.5198]
    • Chrome (web)                 • chrome        •
      web-javascript • Google Chrome 131.0.6778.109
    • Edge (web)                   • edge          •
      web-javascript • Microsoft Edge 131.0.2903.86

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @majid-2002, please provide the full output of `flutter run -v`. 

Did this start after an upgrade or have you always had this issue? If it was after an upgrade, which version of flutter did you upgrade from?

**Comment by Khuza1ma** [comment]
Hi @danagbemava-nc , I’m experiencing the same issue after upgrading Flutter from version 3.24.1 to 3.24.5. I’ve attached the output of flutter run -v in the flutter_logs.txt file for your reference.
[flutter_logs.txt](https://github.com/user-attachments/files/18063430/flutter_logs.txt)

**Comment by danagbemava-nc** [comment]
HI @Khuza1ma, do you get this issue with a newly created flutter app without any plugins?

**Comment by devasya-kaleedo** [comment]
Started facing this issue after doing flutter upgrade just now.

**Comment by danagbemava-nc** [comment]
Hi @Khuza1ma, I can't repro with the plugins/packages you have provided. I think this might be a cache issue. Can you try clearing your gradle cache and running the app again to see if you still experience this issue?

```
Launching lib/main.dart on sdk gphone64 arm64 in debug mode...
Your project is configured with Android NDK 23.1.7779620, but the following plugin(s) depend on a different Android NDK version:
- google_mlkit_barcode_scanning requires Android NDK 25.1.8937393
- google_mlkit_commons requires Android NDK 25.1.8937393
- google_mlkit_digital_ink_recognition requires Android NDK 25.1.8937393
- google_mlkit_entity_extraction requires Android NDK 25.1.8937393
- google_mlkit_face_detection requires Android NDK 25.1.8937393
- google_mlkit_face_mesh_detection requires Android NDK 25.1.8937393
- google_mlkit_image_labeling requires Android NDK 25.1.8937393
- google_mlkit_language_id requires Android NDK 25.1.8937393
- google_mlkit_object_detection requires Android NDK 25.1.8937393
- google_mlkit_pose_detection requires Android NDK 25.1.8937393
- google_mlkit_selfie_segmentation requires Android NDK 25.1.8937393
- google_mlkit_smart_reply requires Android NDK 25.1.8937393
- google_mlkit_text_recognition requires Android NDK 25.1.8937393
- google_mlkit_translation requires Android NDK 25.1.8937393
- shared_preferences_android requires Android NDK 25.1.8937393
Fix this issue by using the highest Android NDK version (they are backward compatible).
Add the following to /Users/deanli/projects/lighthouse/android/app/build.gradle:

    android {
        ndkVersion = "25.1.8937393"
        ...
    }

warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
3 warnings
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
3 warnings
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
3 warnings
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
Note: /Users/deanli/.pub-cache/hosted/pub.dev/google_mlkit_digital_ink_recognition-0.13.0/android/src/main/java/com/google_mlkit_digital_ink_recognition/DigitalInkRecognizer.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
3 warnings
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
Note: /Users/deanli/.pub-cache/hosted/pub.dev/google_mlkit_entity_extraction-0.14.0/android/src/main/java/com/google_mlkit_entity_extraction/EntityExtractor.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
3 warnings
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
Note: /Users/deanli/.pub-cache/hosted/pub.dev/google_mlkit_face_detection-0.12.0/android/src/main/java/com/google_mlkit_face_detection/FaceDetector.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
3 warnings
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
Note: /Users/deanli/.pub-cache/hosted/pub.dev/google_mlkit_face_mesh_detection-0.3.0/android/src/main/java/com/google_mlkit_face_mesh_detection/FaceMeshDetector.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
3 warnings
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
3 warnings
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
Note: /Users/deanli/.pub-cache/hosted/pub.dev/google_mlkit_object_detection-0.14.0/android/src/main/java/com/google_mlkit_object_detection/ObjectDetector.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
3 warnings
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
Note: /Users/deanli/.pub-cache/hosted/pub.dev/google_mlkit_pose_detection-0.13.0/android/src/main/java/com/google_mlkit_pose_detection/PoseDetector.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
3 warnings
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
Note: /Users/deanli/.pub-cache/hosted/pub.dev/google_mlkit_selfie_segmentation-0.9.0/android/src/main/java/com/google_mlkit_selfie_segmentation/SelfieSegmenter.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
3 warnings
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
3 warnings
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
3 warnings
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
3 warnings
Running Gradle task 'assembleDebug'...                             98.3s
✓ Built build/app/outputs/flutter-apk/app-debug.apk
Installing build/app/outputs/flutter-apk/app-debug.apk...           7.4s
Syncing files to device sdk gphone64 arm64...                      742ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h List all available interactive commands.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).

A Dart VM Service on sdk gphone64 arm64 is available at: http://127.0.0.1:52273/fA5ai3zHg0M=/
I/mple.lighthouse( 6405): Compiler allocated 5250KB to compile void android.view.ViewRootImpl.performTraversals()
The Flutter DevTools debugger and profiler on sdk gphone64 arm64 is available at:
http://127.0.0.1:9104?uri=http://127.0.0.1:52273/fA5ai3zHg0M=/
D/ProfileInstaller( 6405): Installing profile for com.example.lighthouse
```

**Comment by majid-2002** [comment]
Hi I was able to solve this issue once. I updated these.

gradle-wrapper.properties
```console
distributionPath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-8.9-all.zip

```
settings.gradle

```console
pluginManagement {
    def flutterSdkPath = {
        def properties = new Properties()
        file("local.properties").withInputStream { properties.load(it) }
        def flutterSdkPath = properties.getProperty("flutter.sdk")
        assert flutterSdkPath != null, "flutter.sdk not set in local.properties"
        return flutterSdkPath
    }()

    includeBuild("$flutterSdkPath/packages/flutter_tools/gradle")

    repositories {
        google()
        mavenCentral()
        gradlePluginPortal()
    }
}

plugins {
    id "dev.flutter.flutter-plugin-loader" version "1.0.0"
    id "com.android.application" version "8.7.0" apply false
    id "org.jetbrains.kotlin.android" version "2.0.21" apply false
}

include ":app"
```



android/app/bulid.gradle

```console
plugins {
    id "com.android.application"
    id "kotlin-android"
    // The Flutter Gradle Plugin must be applied after the Android and Kotlin Gradle plugins.
    id "dev.flutter.flutter-gradle-plugin"
}

android {
    namespace = "com.example.example_app"
    compileSdk = flutter.compileSdkVersion
    ndkVersion = "27.0.12077973"

    compileOptions {
        sourceCompatibility = JavaVersion.VERSION_1_8
        targetCompatibility = JavaVersion.VERSION_1_8
    }

    dependencies {
        coreLibraryDesugaring("com.android.tools:desugar_jdk_libs:2.1.3")
    }
    
    kotlinOptions {
        jvmTarget = JavaVersion.VERSION_1_8
    }

    defaultConfig {
        // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
        applicationId = "com.example.carestack_wellness_app"
        // You can update the following values to match your application needs.
        // For more information, see: https://flutter.dev/to/review-gradle-config.
        minSdk = flutter.minSdkVersion
        targetSdk = flutter.targetSdkVersion
        versionCode = flutter.versionCode
        versionName = flutter.versionName
    }

    buildTypes {
        release {
            // TODO: Add your own signing config for the release build.
            // Signing with the debug keys for now, so `flutter run --release` works.
            signingConfig = signingConfigs.debug
        }
    }
}

flutter {
    source = "../.."
}
```

**Comment by SurajDsDev** [comment]
Can you please provide which JDK version, gradle version and flutter version you are using, where you resolved the above issue.

**Comment by majid-2002** [comment]
I was using gradle 7.6.3  and Flutter 3.24.5

**Comment by darshanshah03** [comment]
My JDK version is 17, with Flutter 3.24.5 and gradle 7.3 then too its not working

**Comment by Yashbhosale00** [comment]
heyy same issue i facing what it the solution ? 
![Screenshot_2023-10-08_11_22_31](https://github.com/user-attachments/assets/1c72339c-047e-4eb0-85b7-587b78524906)

**Comment by Yashbhosale00** [comment]
"I upgraded Flutter, and now an issue is occurring."
![Screenshot_2024-12-11_20_23_24](https://github.com/user-attachments/assets/9c618b4c-cb3e-4e50-949a-242011b8d15c)

Would you like help with resolving the issue you're facing after upgrading Flutter?
any Solution?

**Comment by Khuza1ma** [comment]
Hii @danagbemava-nc 

I did try multiple times deleting the .gradle cache folder and re-running the app but the same issue still persists. I also tried running the app after removing all plugins to rule out any plugin-related issues but that didn’t help either.

Additionally I updated my files according to @majid-2002 solution but unfortunately it didn’t resolve the problem for me.

For reference, I’m using the following setup:

`Flutter: 3.24.5
Dart: 3.5.4
JDK: Java version "17.0.12"`

gradle-wrapper.properties file:

`distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-8.9-all.zip
`
settings.gradle file:

`pluginManagement {
    def flutterSdkPath = {
        def properties = new Properties()
        file("local.properties").withInputStream { properties.load(it) }
        def flutterSdkPath = properties.getProperty("flutter.sdk")
        assert flutterSdkPath != null, "flutter.sdk not set in local.properties"
        return flutterSdkPath
    }()

    includeBuild("$flutterSdkPath/packages/flutter_tools/gradle")

    repositories {
        google()
        mavenCentral()
        gradlePluginPortal()
    }
}

plugins {
    id "dev.flutter.flutter-plugin-loader" version "1.0.0"
    id "com.android.application" version "8.7.3" apply false
    id "org.jetbrains.kotlin.android" version "2.0.21" apply false
}

include ":app"
`
app/build.gradle file:

`plugins {
    id "com.android.application"
    id "kotlin-android"
    // The Flutter Gradle Plugin must be applied after the Android and Kotlin Gradle plugins.
    id "dev.flutter.flutter-gradle-plugin"
}

android {
    namespace = "com.example.savory_chef"
    compileSdk = flutter.compileSdkVersion
    ndkVersion = "27.0.12077973"

    compileOptions {
        sourceCompatibility = JavaVersion.VERSION_17
        targetCompatibility = JavaVersion.VERSION_17
    }

    kotlinOptions {
        jvmTarget = JavaVersion.VERSION_17
    }

    defaultConfig {
        // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
        applicationId = "com.example.savory_chef"
        // You can update the following values to match your application needs.
        // For more information, see: https://flutter.dev/to/review-gradle-config.
        minSdk = flutter.minSdkVersion
        targetSdk = flutter.targetSdkVersion
        versionCode = flutter.versionCode
        versionName = flutter.versionName
    }

    buildTypes {
        release {
            // TODO: Add your own signing config for the release build.
            // Signing with the debug keys for now, so `flutter run --release` works.
            signingConfig = signingConfigs.debug
        }
    }
}

flutter {
    source = "../.."
}
`

**Comment by majid-2002** [comment]
You should update jdk to 21 . The issue persists with jdk 17

**Comment by Khuza1ma** [comment]
Hey @majid-2002 

Flutter currently works best with JDK 17 as it’s the most stable and supported version. JDK 21 might seem like a fix but it’s not fully compatible with Flutter yet and could cause more issues I guess.

**Comment by majid-2002** [comment]
> Hey @majid-2002 
> 
> Flutter currently works best with JDK 17 as it’s the most stable and supported version. JDK 21 might seem like a fix but it’s not fully compatible with Flutter yet and could cause more issues I guess.

Hey latest flutter stable version works on jdk 21.

I think you might need to downgrade flutter version. If you are running jdk 17.

As you can see in flutter doctor -v it clearly says it runs on jdk 21

**Comment by danagbemava-nc** [comment]
Hi @Khuza1ma, is there a particular issue you're experiencing with jdk 21? The JdkImageTransforms issue which was introduced when upgrading to the latest android studio was due to an incompatibility between gradle and that version of the jdk. With the gradle versions you have specified, you should not have that issue

**Comment by darshanshah03** [comment]
Hey @majid-2002 

I used the same version stack which you have given

Flutter: 3.24.5
JDK: 21
Gradle: 7.6.3

facing below error

![Screenshot 2024-12-12 165635](https://github.com/user-attachments/assets/53c661d6-766d-4d49-9685-cf8cc4d3132a)

**Comment by majid-2002** [comment]
> Hey @majid-2002 
> 
> I used the same version stack which you have given
> 
> Flutter: 3.24.5
> JDK: 21
> Gradle: 7.6.3
> 
> facing below error
> 
> ![Screenshot 2024-12-12 165635](https://github.com/user-attachments/assets/53c661d6-766d-4d49-9685-cf8cc4d3132a)
> 

Hey , try updating gradle to 8.9. 
It resolved for me.

**Comment by Khuza1ma** [comment]
Hi @danagbemava-nc ,

Thanks for following up! The issue I was experiencing is now resolved after upgrading to the latest Flutter version (3.27.0) released yesterday. It seems the newer version addressed the pom file issue I was facing.

Appreciate the help!

**Comment by danagbemava-nc** [comment]
Closing this since the original reporter has resolved their issue and it appears that this can be resolved by upgrading to 3.27.0. 

If you are using gradle 8 and the latest version of flutter stable (3.27.0) but still encounter this issue, kindly file a new issue and provide the relevant information required to reproduce the issue. 

Thank you
